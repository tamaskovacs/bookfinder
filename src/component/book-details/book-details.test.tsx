import React from 'react';
import { render } from '@testing-library/react';
import BookDetails from './book-details';

describe('<BookDetails /> ', () => {
    it('should match the snapshot if has bookdata', () => {
        const bookData = {
            title: 'test title',
            author_name: 'author name',
            publish_date: ['2222 jan 1', '3333 dec 12'],
            publisher: ['any publisher', 'other publisher'],
            subject: ['any subject', 'other subject']
        };

        const elementFilled = render(<BookDetails data={bookData} />);
        expect(elementFilled).toMatchSnapshot();
    });

    it('should match the snapshot has not bookdata', () => {
        const bookData = {
            title: '',
            author_name: '',
            publish_date: [],
            publisher: [],
            subject: []
        };
        const elementUnFilled = render(<BookDetails data={bookData} />);
        expect(elementUnFilled).toMatchSnapshot();
    });

});

