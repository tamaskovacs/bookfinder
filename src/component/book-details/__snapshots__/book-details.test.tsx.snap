// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetails />  should match the snapshot has not bookdata 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="table"
      >
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Title
          </div>
          <div
            class="book-cell"
          />
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Author
          </div>
          <div
            class="book-cell"
          />
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Publish date
          </div>
          <div
            class="book-cell"
          />
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Publisher
          </div>
          <div
            class="book-cell"
          />
        </div>
        <div
          class="book-row"
          id="book-row-1"
        >
          <div
            class="param-cell"
          >
            Subject
          </div>
          <div
            class="book-cell"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="table"
    >
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Title
        </div>
        <div
          class="book-cell"
        />
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Author
        </div>
        <div
          class="book-cell"
        />
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Publish date
        </div>
        <div
          class="book-cell"
        />
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Publisher
        </div>
        <div
          class="book-cell"
        />
      </div>
      <div
        class="book-row"
        id="book-row-1"
      >
        <div
          class="param-cell"
        >
          Subject
        </div>
        <div
          class="book-cell"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<BookDetails />  should match the snapshot if has bookdata 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="table"
      >
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Title
          </div>
          <div
            class="book-cell"
          >
            test title
          </div>
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Author
          </div>
          <div
            class="book-cell"
          >
            author name
          </div>
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Publish date
          </div>
          <div
            class="book-cell"
          >
            2222 jan 1, 3333 dec 12
          </div>
        </div>
        <div
          class="book-row"
        >
          <div
            class="param-cell"
          >
            Publisher
          </div>
          <div
            class="book-cell"
          >
            any publisher, other publisher
          </div>
        </div>
        <div
          class="book-row"
          id="book-row-1"
        >
          <div
            class="param-cell"
          >
            Subject
          </div>
          <div
            class="book-cell"
          >
            any subject, other subject
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="table"
    >
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Title
        </div>
        <div
          class="book-cell"
        >
          test title
        </div>
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Author
        </div>
        <div
          class="book-cell"
        >
          author name
        </div>
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Publish date
        </div>
        <div
          class="book-cell"
        >
          2222 jan 1, 3333 dec 12
        </div>
      </div>
      <div
        class="book-row"
      >
        <div
          class="param-cell"
        >
          Publisher
        </div>
        <div
          class="book-cell"
        >
          any publisher, other publisher
        </div>
      </div>
      <div
        class="book-row"
        id="book-row-1"
      >
        <div
          class="param-cell"
        >
          Subject
        </div>
        <div
          class="book-cell"
        >
          any subject, other subject
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
